dtlpyconverters/vtt/vtt_converters.py

```diff
@@ -0,0 +1,290 @@
+import json
+import os
+import asyncio
+import nest_asyncio
+
+from ..base import BaseExportConverter, BaseImportConverter
+
+logger = logging.getLogger(name='dtlpy')
+
+
+class VttToDataloop(BaseImportConverter):
+    def __init__(self,
+                 dataset: dl.Dataset,
+                 input_annotations_path,
+                 output_annotations_path=None,
+                 input_items_path=None,
+                 upload_items=False,
+                 add_labels_to_recipe=True,
+                 concurrency=6,
+                 return_error_filepath=False,
+                 ):
+
+        nest_asyncio.apply()
+
+        # global vars
+        super(VttToDataloop, self).__init__(
+            dataset=dataset,
+            output_annotations_path=output_annotations_path,
+            input_annotations_path=input_annotations_path,
+            input_items_path=input_items_path,
+            upload_items=upload_items,
+            add_labels_to_recipe=add_labels_to_recipe,
+            concurrency=concurrency,
+            return_error_filepath=return_error_filepath,
+        )
+
+    @staticmethod
+    def _get_event_loop():
+        try:
+            loop = asyncio.get_event_loop()
+        except RuntimeError as e:
+            if "no current event loop" in str(e):
+                loop = asyncio.new_event_loop()
+                asyncio.set_event_loop(loop)
+            else:
+                raise e
+        return loop
+
+    async def convert_dataset(self,
+                              speaker_in_caption=True,
+                              speaker_seperator=':',
+                              speaker_to_annotation=True):
+        self.speaker_in_caption = speaker_in_caption
+        self.speaker_seperator = speaker_seperator
+        self.speaker_to_annotation = speaker_to_annotation
+        # inputs
+        # read labels and handle recipes
+        # with open(self.label_txt_filepath, 'r') as f:
+        #     self.id_to_label_map = {i_label: label.strip() for i_label, label in enumerate(f.readlines())}
+        # if self.add_labels_to_recipe:
+        #     self.dataset.add_labels(label_list=list(self.id_to_label_map.values()))
+
+        # read annotations files and run on items
+        files = list(Path(self.input_annotations_path).rglob('*.vtt'))
+        files.extend(list(Path(self.input_annotations_path).rglob('*.rst')))
+
+        tasks = [self.on_item(annotation_filepath=str(txt_file)) for txt_file in files]
+
+        # Running the tasks concurrently
+        await asyncio.gather(*tasks)
+
+    async def on_item(self, **context):
+        """
+
+        """
+
+        annotation_filepath = context.get('annotation_filepath')
+
+        # find images with the same name (ignore image ext)
+        relpath = os.path.relpath(annotation_filepath, self.input_annotations_path)
+        filename, ext = os.path.splitext(relpath)
+        audio_filepaths = list(Path(os.path.join(self.input_items_path)).rglob(f'{filename}.*'))
```

Comment:
If self.input_annotations_path == self.input_items_path 
it will not work.
We don't want to filter only files of audio type?